activerecord/lib/active_record/inheritance.rb

```diff
@@ -67,10 +67,14 @@ def new(attributes = nil, &block)
         end
 
         if subclass && subclass != self
-          subclass.new(attributes, &block)
-        else
-          super
+          return subclass.new(attributes, &block)
         end
+
+        if abstract_class?
```

Comment:
I think it's clearer if this check is an `elsif` for  `if _has_attribute?(inheritance_column)`.
This check fails if the inheritance_column attribute is empty and it's a abstract_class.
```
if _has_attribute?(inheritance_column)
   ...
elsif abstract_class?
  ...
end
```

Reply:
I committed the change and ran the test, but it failed and it made me realize this isn't want we want.

The issue here is to prevent `super` to be called if this is an abstract class. One possible alternative to this would be the following:

```
if subclass && subclass != self
  subclass.new(attributes, &block)
elsif abstract_class?
  raise NotImplementedError, "#{self} is an abstract class and cannot be instantiated without setting the '#{inheritance_column}' attribute."
else
  super
end
```

If I move the `elsif abstrac_class?` above line 70, it [breaks a test that were existing before this PR](https://github.com/rails/rails/blob/98a7e0747f98a1dfd71989c5e500d1b06ebbe9b8/activerecord/test/cases/inheritance_test.rb#L321-L326).

Reply:
I also rebased, and forced push, and the build didn't restart. Do I need to do an "empty" commit to reforce a build?

Reply:
I think a force push should work..

Reply:
Hi @pothibo,
Sorry for the late reply.
I think the current version is failing because of the missing `super` at the bottom.
The implementation you posted above would probably fix this PR:
```
if subclass && subclass != self
  subclass.new(attributes, &block)
elsif abstract_class?
  raise NotImplementedError, "#{self} is an abstract class and cannot be instantiated without setting the '#{inheritance_column}' attribute."
else
  super
end
```